### ISSUE TITLE
Duplicate logic for closing?

### ISSUE THREAD

**Issue by staltz** [ssbc/ssb-gossip#5]
As I'm rewriting the gossip plugin and studying it, I stumbled upon what looks like duplicate code:

`ssb-gossip`>`index.js` has a `server.close` hook to run close on each peer:

https://github.com/ssbc/ssb-gossip/blob/5452a2625befad1ab7ae5e3e047df2f27f7b02fa/index.js#L128-L131

But so does secret-stack, because `server.peers` above is built by secret-stack, and here we see in `secret-stack`>`core.js`:

https://github.com/ssbc/secret-stack/blob/16eb4d9b2ff7da3ac5ff89f616f38fcd5a4666f3/core.js#L246-L252

So I believe it is safe to assume we can delete the close logic in the gossip plugin.

**Comment by dominictarr** [comment]
yes, we should have secret-stack handle that. also, originally, it was intended that close didn't fully close until everything was finished, but that is actually a pain for testing. close should just kill all connections immediately.
